### PR TITLE
Remove discontinued device types from generated libraries

### PR DESCRIPTION
**Pull request by klutchell**

### CODE REVIEW

**Comment by klutchell** [comment]
converting to draft until I squash the commits

File: scripts/generate-dockerhub-description.js [hunk 1]
```diff
@@ -119,6 +128,27 @@ if (types.indexOf('all') > -1) {
...
+						type: 'string',
+						enum: supportedDeviceTypes.map((device) => device.slug),
+					},
+				},
+			};
+
+			if (['stack-device', 'os-device'].includes(type)) {
+				query.selector = {
+					...query.selector,
+					['hw.device-type']: {
```
Comment (pipex) [inline scripts/generate-dockerhub-description.js:145]: This is shorter
```suggestion
				query.selector['hw.device-type'] = {
```

File: scripts/generate-dockerhub-description.js [hunk 2]
```diff
@@ -119,6 +128,27 @@ if (types.indexOf('all') > -1) {
 
 			const query = yaml.load(await fs.readFile(BLUEPRINT_PATHS[type], 'utf8'));
 
+			// Remove discontinued device types
+			const supportedDeviceFilter = {
```
Comment (pipex) [inline scripts/generate-dockerhub-description.js:132]: You can put this  outside of the loop, since you only need to define it once

**Comment by pipex** [comment]
LGTM

**Pull request merged**